### PR TITLE
Offer Reset: add Jetpack CRM to Plans page

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* Include Jetpack CRM in the Plans page.

#### Implementation notes

* Jetpack CRM is not well supported from inside Calypso. We don't seem to have a way to fetch its price or to know if a site has it. Because of this, I categorized the product as an external product. My goal was to implement this in a way that will allows us a smooth transition into a real product once our endpoints return the data we need for that and without disrupting our current logic by adding conditionals everywhere.

#### Testing instructions

* Download this PR and run both Calypso Blue and Calypso Green in parallel with `yarn start` and `yarn start-jetpack-cloud-p` respectively. `yarn start-jetpack-cloud-p` must be executed after `yarn start` has finished.

**Pricing page**

* Visit http://jetpack.cloud.localhost:3001/pricing.
* Make sure that when you select `All` in the Product Type filter, Jetpack CRM card appears just below Jetpack Search.

**WordPress.com Plans**
* Visit http://calypso.localhost:3000/plans/:site.
* Make sure that when you select `All` in the Product Type filter, Jetpack CRM card appears just below Jetpack Search.

**Connect Flow**
* Visit http://calypso.localhost:3000/jetpack/connect/store.
* Make sure that when you select `All` in the Product Type filter, Jetpack CRM card appears just below Jetpack Search.

**All cases**
* Make sure the copy is correct (check the Jetpack Offer Reset Copy linked in our board).
* Make sure that the feature list is correct (also defined in the same document).
* Make sure that the CTA takes users to https://jetpackcrm.com/pricing/.
* Make sure you don't see CRM when the monthly term is selected (CRM is billed annually)
* Make sure that when you own Jetpack Complete, you see the _Included in your plan_ mention

Fixes 1169247016322522-as-1194149699265809

#### Demo

A new icon has been added since these demos were done.

##### WordPress.com Plans
image

##### Jetpack Connect
image

##### Jetpack cloud
image

### CODE REVIEW

File: client/my-sites/plans-v2/product-card/index.tsx [hunk 1]
```diff
@@ -148,6 +153,13 @@ const ProductCardWrapper = ( {
 	const isUpgradeableToYearly =
 		isOwned && selectedTerm === TERM_ANNUALLY && item.term === TERM_MONTHLY;
 
+	// Don't hide price if siteId is not defined, since it most likely won't be shown
+	// in other parts of the card (e.g. Jetpack Search)
+	// Jetpack CRM is an external product and we don't have access to its price, therefore,
+	// we hide it.
+	const hidePrice =
+		EXTERNAL_PRODUCTS_LIST.includes( item.productSlug ) || ( !! siteId && item.hidePrice );
```
Comment (rcanepa) [inline client/my-sites/plans-v2/product-card/index.tsx:161]: I'm not very pleased about this. I think we might have to refactor it, but no sure if this is the right PR to do that.

**Review by mattgawarecki — APPROVED** [review]
I tested every case in the PR description and it all behaved as expected. I agree there are some things for us to come back and polish once CRM is more thoroughly integrated, but the code looks good even so! 🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4451174

Thank you @rcanepa for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4451174).

**Pull request merged**